gmail/src/views/create_task.ts

```diff
@@ -0,0 +1,113 @@
+import { buildView } from "../views/index";
+import { updateCard, pushToRoot } from "./helpers";
+import { UI_ICONS } from "./icons";
+import { createKeyValueWidget, actionCall, notify } from "./helpers";
+import { URLS } from "../const";
+import { ErrorMessage } from "../models/error_message";
+import { Project } from "../models/project";
+import { State } from "../models/state";
+import { Task } from "../models/task";
+import { logEmail } from "../services/log_email";
+import { _t } from "../services/translation";
+
+function onSearchProjectClick(state: State, parameters: any, inputs: any) {
+    const inputQuery = inputs.search_project_query;
+    const query = (inputQuery && inputQuery.length && inputQuery[0]) || "";
+    const [projects, error] = Project.searchProject(query);
+
+    state.error = error;
+    state.searchedProjects = projects;
+
+    return updateCard(buildCreateTaskView(state, query));
+}
+
+function onCreateProjectClick(state: State, parameters: any, inputs: any) {
+    const inputQuery = inputs.new_project_name;
+    const projectName = (inputQuery && inputQuery.length && inputQuery[0]) || "";
+
+    if (!projectName || !projectName.length) {
+        return notify(_t("The project name is required"));
+    }
+
+    const project = Project.createProject(projectName);
+    if (!project) {
+        return notify(_t("Could not create the project"));
+    }
+
+    return onSelectProject(state, { project: project });
+}
+
+function onSelectProject(state: State, parameters: any) {
+    const project = Project.fromJson(parameters.project);
+    const task = Task.createTask(state.partner.id, project.id, state.email.body, state.email.subject);
```

Comment:
By the way, what happens if email has no subject ? Do we set a default one ?

Reply:
It will create a task with an empty name
![screenshot_spect](https://user-images.githubusercontent.com/54316226/125923828-78552237-5c98-411b-aa48-d179f2c3969e.png)

But the body of the email will be added in the description, not sure what we should do in that case

Reply:
Not really nice. But probably belong to a side task, to check with LNA, as this impacts also helpdesk.